### ISSUE TITLE
How to training with EMA in model.train()?

### ISSUE THREAD

**Issue by htwang14** [ultralytics/ultralytics#16329]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I trained the yolo model using `model.train(data="coco8.yaml", epochs=100, imgsz=640)` and everything worked fine. I suppose the above code is not applying EMA, right? Could you please show me how to apply EMA during training using the `model.train` function?

I've checked your code demo [here](https://github.com/ultralytics/ultralytics/issues/8216#issuecomment-1946401506). But it is not implemented with the `model.train`. Thank you in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @htwang14, thank you for reaching out and for your interest in Ultralytics 🚀! We appreciate you bringing up this topic. This is an automated response to assist you promptly, and an Ultralytics engineer will follow up soon.

If you're inquiring about training enhancements like EMA, please ensure you're checking out our [Docs](https://docs.ultralytics.com) for guidance and examples on model training.

In case this is part of a custom training ❓ question, do provide as much detail as possible, such as dataset snippets and any logs, for more effective assistance. And ensure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our Ultralytics community to engage with more in-depth discussions:
- Real-time support on [Discord](https://ultralytics.com/discord) 🎧
- Participate in discussions on [Discourse](https://community.ultralytics.com)
- Explore threads or share insights on our [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Ensure you’re using the latest package version to benefit from the newest features:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLOv8 in these verified environments with preinstalled dependencies:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Services**: [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates that all tests are passing for the day’s CI checks. For further details on YOLOv8's [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/), please reference our documentation.

Stay tuned, and our team will assist with more details on implementing EMA in your training process! 😊

**Comment by Y-T-G** [comment]
Ultralytics uses EMA by default.

https://github.com/ultralytics/ultralytics/blob/c5bee7545cb5fe26e201c593f3920b4b462c73c1/ultralytics/engine/trainer.py#L300
